elasticsearch/includes/classes/class-elasticsearch-integration.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Automattic\VIP\Elasticsearch;
+
+class Integration {
+
+	function __construct() {
+		add_filter( 'ep_skip_query_integration', array( __CLASS__, 'ep_skip_query_integration' ) );
+	}
+
+	/**
+	 * Separate plugin enabled and querying the index
+	 *
+	 * The index can be tested at any time by setting an `es` query argument.
+	 * When we're ready to use the index in production, the `vip_enable_elasticsearch`
+	 * option will be set to `true`, which will enable querying for everyone.
+	 */
+	function ep_skip_query_integration() {
+		if ( isset( $_GET[ 'es' ] ) ) {
+			return false;
+		}
+
+		if ( get_option( 'vip_enable_elasticsearch' ) ) {
```

Comment:
I think an option is too flimsy for this - a constant would be better, because an option can be deleted / changed w/o an audit trail, or the caching could be broken, etc, which could have significant repercussions in production.

Related - we should rename the `USE_VIP_ELASTICSEARCH` constant too, to complement whatever we decide on here.

Reply:
> I think an option is too flimsy for this - a constant would be better, because an option can be deleted / changed w/o an audit trail, or the caching could be broken, etc, which could have significant repercussions in production.

This seems like it would also work well when we have a UI on the VIP Dashboard for Environment Variables, what do you think?